### PR TITLE
[CPDNPQ-2270] port rake file and service from ECF (bulk script for revert to pending)

### PR DESCRIPTION
**Pull request by alkesh**
### Context

Ticket: https://dfedigital.atlassian.net/browse/CPDNPQ-2270

bulk script for revert to pending - rake file and service ported from ECF - updated to work with the NPQ `Applications::RevertToPending` service.

### CODE REVIEW

File: lib/tasks/bulk_change_applications_to_pending.rake [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    dry_run = args[:dry_run] != "false"
+    unless File.exist?(csv_file_path)
+      logger.error "File not found: #{csv_file_path}"
+      return
+    end
+
+    application_ecf_ids = CSV.read(csv_file_path, headers: false).flatten
+
+    logger.info "Bulk changing #{application_ecf_ids.size} applications to pending#{' (dry run)' if dry_run}..."
+
```
Comment (jebw) [inline lib/tasks/bulk_change_applications_to_pending.rake:30]: `sleep(10)` - gives the developer time to think 'oops' and press ctrl-c ? Might be belt and braces though, there's already a dry run feature? thoughts?

File: spec/lib/tasks/one_off/bulk_change_applications_to_pending_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    subject { Rake::Task["one_off:bulk_change_to_pending"].invoke(csv_file.path) }
+
+    it "does not update any applications" do
+      subject
+      expect(Application.all.pluck(:lead_provider_approval_status)).to eq %w[accepted accepted]
+    end
+  end
+
+  context "when dry run true" do
+    subject { Rake::Task["one_off:bulk_change_to_pending"].invoke(csv_file.path, "true") }
```
Comment (jebw) [inline spec/lib/tasks/one_off/bulk_change_applications_to_pending_spec.rb:29]: `subject(:run_task)`

File: spec/lib/tasks/one_off/bulk_change_applications_to_pending_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    it "does not update any applications" do
+      subject
```
Comment (jebw) [inline spec/lib/tasks/one_off/bulk_change_applications_to_pending_spec.rb:32]: `run_task` ? Clarifies what subject is doing ?

**Review by jebw — APPROVED** [review]
Couple of minor code clarity suggestions but looks great :raised_hands:

File: lib/tasks/bulk_change_applications_to_pending.rake (hunk 1 above)
Comment (alkesh) [reply lib/tasks/bulk_change_applications_to_pending.rake:30]: I think this one might be overkill.
I'd expect the developer will already run this with dry-run first, and the effort in creating/uploading the CSV file in the first place means it shouldn't be a trivial thing to run accidentally.

**Pull request merged**